Kernel/Net/LocalSocket.cpp

```diff
@@ -399,14 +399,27 @@ ErrorOr<void> LocalSocket::getsockopt(OpenFileDescription& description, int leve
 
     MutexLocker locker(mutex());
 
+    size_t buffer_size;
     socklen_t size;
     TRY(copy_from_user(&size, value_size.unsafe_userspace_ptr()));
 
     switch (option) {
     case SO_SNDBUF:
-        return ENOTSUP;
+        if (size < sizeof(size_t))
+            return EINVAL;
+        buffer_size = m_for_server.ptr()->space_for_writing();
+        TRY(copy_to_user(static_ptr_cast<int*>(value), bit_cast<int*>(&buffer_size)));
```

Comment:
Again, the pointer cast is utterly wrong and the type is size_t. This bit_cast is wrong and will be UB on big-endian systems.

Reply:
what cast should be used here? pointer cast seems to be used on everywhere else